contracts/libraries/OracleLibrary.sol

```diff
@@ -76,4 +76,34 @@ library OracleLibrary {
 
         return uint32(block.timestamp) - observationTimestamp;
     }
+
+    /// @notice Given a pool, it returns the tick value as of the start of the current block
+    /// @param pool Address of Uniswap V3 pool
+    /// @return The tick that the pool was in at the start of the current block
+    function getBlockStartingTick(address pool) internal view returns (int24) {
+        (, int24 tick, uint16 observationIndex, uint16 observationCardinality, , , ) = IUniswapV3Pool(pool).slot0();
+
+        // 3 observations are needed to reliably calculate the block starting tick
+        require(observationCardinality > 2, 'NEO');
```

Comment:
we don't need this condition here!

in the next block, you (correctly) check the timestamp of the current observation, and if it's in the past, safely return the current tick as the block-starting tick. note that this relies only on a single observation.

you do need a cardinality check _after_ that block, but it should be > 1, not > 2.